src/applications/personalization/preferences/actions/index.js

```diff
@@ -1,33 +1,138 @@
-import localStorage from 'platform/utilities/storage/localStorage';
+import { apiRequest } from 'platform/utilities/api';
 
-export const DASHBOARD_PREFERENCE_SET = 'DASHBOARD_PREFERENCE_SET';
-export const DASHBOARD_PREFERENCES_SAVED = 'DASHBOARD_PREFERENCES_SAVED';
-export const DASHBOARD_PREFERENCES_FETCHED = 'DASHBOARD_PREFERENCES_FETCHED';
+import { LOADING_STATES, PREFERENCE_CODES } from '../constants';
+import { transformPreferencesForSaving } from '../helpers';
 
-export function fetchPreferences() {
-  const savedPrefs = localStorage.getItem('dashboard-preferences');
-  localStorage.setItem('dashboardLastVisitedAt', Date.now());
+export const SET_USER_PREFERENCE_REQUEST_STATUS =
+  'SET_USER_PREFERENCE_REQUEST_STATUS';
+export const SET_ALL_PREFERENCE_OPTIONS_REQUEST_STATUS =
+  'SET_ALL_PREFERENCE_OPTIONS_REQUEST_STATUS';
+export const SET_SAVE_PREFERENCES_REQUEST_STATUS =
+  'SET_SAVE_PREFERENCES_REQUEST_STATUS';
+export const SET_AVAILABLE_BENEFITS = 'SET_AVAILABLE_BENEFITS';
+export const SET_DASHBOARD_USER_PREFERENCES = 'SET_DASHBOARD_USER_PREFERENCES';
+export const SET_DASHBOARD_PREFERENCE = 'SET_DASHBOARD_PREFERENCE';
+export const SAVED_DASHBOARD_PREFERENCES = 'SAVED_DASHBOARD_PREFERENCES';
 
-  return {
-    type: DASHBOARD_PREFERENCES_FETCHED,
-    data: (savedPrefs && JSON.parse(savedPrefs)) || {},
+// load the benefits the user has picked to learn more about
+export function fetchUserSelectedBenefits() {
+  return dispatch => {
+    dispatch({
+      type: SET_USER_PREFERENCE_REQUEST_STATUS,
+      status: LOADING_STATES.pending,
+    });
+    return apiRequest(
+      '/user/preferences',
+      null,
+      response => {
```

Comment:
Recommend moving everything but the dispatch into the reducer so it can be unit tested. I think actions should just dispatch- not mutate, select, query, etc.